### ISSUE TITLE
Accepts invalid dates without warning

### ISSUE THREAD

**Issue by mikerodrigues** [sxywu/expenses#4]
I submitted an expense for something purchased February 31st. It accepted the submission without warning.

It looks like it accepts 1-31 regardless of the month. A calendar picker thing might be better.

**Comment by sxywu** [comment]
@mikerodrigues I agree, and I briefly tried to find a good calendar picker (or even a date validator) last night but didn't come across one that stood out.  I would actually prefer date validation, and I thought d3 put something out.  Maybe I'm wrong.
